src/main/java/com/google/sps/servlets/RecipeServlet.java

```diff
@@ -77,11 +77,37 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     DocumentReference recipeRef = DBReferences.recipes().document();
     newRecipe.id = recipeRef.getId();
-    recipeRef.set(newRecipe);
+    ApiFuture future = recipeRef.set(newRecipe);
+    try{
+      future.get();
+    } catch (InterruptedException e) {
+      System.out.println("Attempt to add recipe raised exception: " + e);
+    } catch (ExecutionException e) {
+      System.out.println("Attempt to add recipe raised exception: " + e);
+    }
 
     response.setContentType("application/json");
     response.getWriter().println(gson.toJson(new DBObject(newRecipe.id)));
-    response.setStatus(HttpServletResponse.SC_ACCEPTED);
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String data = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+    Recipe newRecipe = gson.fromJson(data, Recipe.class);
```

Comment:
Where will this be fetched on the frontend? Are we planning to create a separate page to submit recipe edits?

Reply:
I would think so

Reply:
I think we can include something on the profile page that displays recipes a user has submitted (by redirecting to recipe-display), and put the Edit option on those pages